sdk/python/kfp/dsl/_pipeline_volume.py

```diff
@@ -0,0 +1,407 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from . import _pipeline
+from . import _pipeline_param
+from . import _pipeline_vsnapshot
+from ..compiler._k8s_helper import K8sHelper
+from kubernetes import client as k8s_client
+import re
+from typing import List, Dict
+
+
+VOLUME_MODE_RWO = ["ReadWriteOnce"]
+VOLUME_MODE_RWM = ["ReadWriteMany"]
+VOLUME_MODE_ROM = ["ReadOnlyMany"]
+
+
+class PipelineVolume(object):
+    """Representing a volume that is passed between pipeline operators.
+    That volume could be either new (created according to the user's needs from
+    scratch or from a VolumeSnapshot) or refering to an existing pvc.
+
+    A PipelineVolume object can be used as an extention of the pipeline
+    function's filesystem. It may then be passed between pipeline functions,
+    exposing dependencies.
+    """
+    def __init__(self, name: str = None, pvc: str = None,
+                 size: str = None, storage_class: str = None,
+                 data_source=None, mode: List[str] = VOLUME_MODE_RWM,
+                 annotations: Dict = None,
+                 k8s_resource: k8s_client.V1PersistentVolumeClaim = None,
+                 create: bool = True, is_exit_handler: bool = False):
+        """Create a new instance of PipelineVolume.
+        Args:
+            name: A desired name to refer to the PipelineVolume, and to use as
+                base when naming the created resource. This might change when
+                the pipeline gets compiled, in case the name is already added.
+            pvc: The name of an existing PVC
+            size: The size of the PVC which will be created
+            storage_class: The storage class to use for the dynamically created
+                PVC (requires size or data_source)
+            data_source: The name of an existing VolumeSnapshot or a
```

Comment:
Could you add that it is an alpha feature and requires to be enabled in k8s cluster?

Reply:
Yup! Will add a comment in the final PR.